Riot/Modules/Room/RoomViewController.m

```diff
@@ -1126,7 +1126,11 @@ - (void)updateRoomInputToolbarViewClassIfNeeded
     {
         [super setRoomInputToolbarViewClass:roomInputToolbarViewClass];
         
-        [(RoomInputToolbarView *)self.inputToolbarView setVoiceMessageToolbarView:self.voiceMessageController.voiceMessageToolbarView];
+        // The voice message toolbar cannot be set on DisabledInputToolbarView.
+        if ([self.inputToolbarView isKindOfClass:RoomInputToolbarView.class])
+        {
+            [(RoomInputToolbarView *)self.inputToolbarView setVoiceMessageToolbarView:self.voiceMessageController.voiceMessageToolbarView];
+        }
```

Comment:
Would `respondsToSelector:` and `performSelector:withObject` be a better solution to this?
I've not used those before, but can't see the advantage in this instance given that the method is only defined on `RoomInputToolbarView` and no other types.

Reply:
It is better as you did.
If there is a change in `setVoiceMessageToolbarView` signature, the compiler will complain whereas it will silently fail with `respondsToSelector:` and `performSelector:withObject`.